Lib/ast.py

```diff
@@ -1051,7 +1051,8 @@ def visit_ClassDef(self, node):
             self.fill("@")
             self.traverse(deco)
         self.fill("class " + node.name)
-        self._type_params_helper(node.type_params)
+        if hasattr(node, "type_params"):
```

Comment:
I assume it would be harder to just make sure that these nodes always have a type_params attribute, even if the relevant argument wasn't passed to the constructor?

Reply:
Yes, that might be something to pursue for 3.13 though.

Reply:
#105858